README.md
# Avatar with name initials generated

[![Build Status](https://travis-ci.org/aurostudio/auro-avatar.svg?branch=master)](https://travis-ci.org/aurostudio/auro-avatar)
[![pub package](https://img.shields.io/pub/v/auro_avatar.svg)](https://pub.dartlang.org/packages/auro_avatar)

A flutter plugin to generate an avatar with name initials from a full-name string.

## How to import

Add this to your package's pubspec.yaml file:
```
dependencies:
  auro_avatar: "^0.1.0"

```
Add it to your dart file:
```
import 'package:auro_avatar/auro_avatar.dart';
```
## Usage
The auro_avatar can be used directly as:

#### InitialNameAvatar - Circle avatar

```
new InitialNameAvatar(
    'John Doe',
    circleAvatar: true,
    borderColor: Colors.grey,
    borderSize: 4.0,
    backgroundColor: Colors.blue,
    foregroundColor: Colors.white,
    padding: 20.0,
    textSize: 30.0,
),
 ```

#### InitialNameAvatar - Square avatar

```
new InitialNameAvatar(
    'John Doe',
    circleAvatar: false,
    ...
),
 ```


CHANGELOG.md
## [0.1.1] - 2018-09-22

* Configured travis-ci.

## [0.1.0] - 2018-09-22

Initial release:
* Added the feature to generate an avatar with name initials.


test/widget_test.dart
import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';

import 'package:auro_avatar/auro_avatar.dart';

void main() {
  testWidgets('Verify InitialNameAvatar widget', (WidgetTester tester) async {
    // Build our app and trigger a frame.
    await tester.pumpWidget(
      StatefulBuilder(
        builder: (BuildContext context, StateSetter setState) {
          return MaterialApp(
            home: Material(
              child: Center(
                child: new Column(
                  mainAxisAlignment: MainAxisAlignment.center,
                  children: <Widget>[
                    Container(
                      child: new InitialNameAvatar(
                        'John Doe',
                        circleAvatar: true,
                        borderColor: Colors.grey,
                        borderSize: 4.0,
                        backgroundColor: Colors.blue,
                        foregroundColor: Colors.white,
                        padding: 20.0,
                        textSize: 30.0,
                      ),
                    ),
                  ],
                ),
              ),
            ),
          );
        },
      ),
    );

    // Verify that platform version is retrieved.
    expect(
        find.byWidgetPredicate(
          (Widget widget) =>
              widget is InitialNameAvatar && widget.fullname.startsWith('John'),
        ),
        findsOneWidget);
  });
}


lib/src/initialNameAvatar.dart
import 'package:flutter/material.dart';
import 'package:flutter/widgets.dart';

import 'utils.dart';

class InitialNameAvatar extends StatelessWidget {
  final String fullname;
  final Color borderColor;
  final Color backgroundColor;
  final Color foregroundColor;
  final double textSize;
  final double borderSize;
  final double padding;
  final bool circleAvatar;

  const InitialNameAvatar(
    this.fullname, {
    Key key,
    this.borderColor,
    this.borderSize,
    this.backgroundColor,
    this.foregroundColor,
    this.textSize,
    this.padding,
    this.circleAvatar,
  }) : assert(fullname != null);

  Widget renderCircleAvatar() {
    return new Container(
      decoration: BoxDecoration(
          borderRadius: BorderRadius.all(Radius.circular(50.0)),
          border: Border.all(
            color: this.borderColor != null
                ? this.borderColor
                : Colors.transparent,
            width: this.borderSize != null ? this.borderSize : 2.0,
          )),
      child: CircleAvatar(
        backgroundColor: this.backgroundColor,
        radius: 40.0,
        child: Text(
          getInitials(this.fullname),
          style: TextStyle(
              color: this.foregroundColor,
              fontSize:
                  (this.textSize != null) ? this.textSize : this.textSize),
        ),
      ),
    );
  }

  Widget renderSquareAvatar() {
    return new Container(
      padding: EdgeInsets.all(this.padding != null ? this.padding : 5.0),
      decoration: BoxDecoration(
        color: this.backgroundColor,
        border: Border.all(
          color:
              this.borderColor != null ? this.borderColor : Colors.transparent,
          width: this.borderSize != null ? this.borderSize : 2.0,
        ),
      ),
      child: Text(
        getInitials(this.fullname),
        style: TextStyle(
            color: this.foregroundColor,
            fontSize: (this.textSize != null) ? this.textSize : this.textSize),
      ),
    );
  }

  Widget build(BuildContext context) {
    return this.circleAvatar ? renderCircleAvatar() : renderSquareAvatar();
  }
}


android/src/main/java/com/aurostudio/flutter/auroavatar/AuroAvatarPlugin.java
package com.aurostudio.flutter.auroavatar;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;
import io.flutter.plugin.common.PluginRegistry.Registrar;

/** AuroAvatarPlugin */
public class AuroAvatarPlugin implements MethodCallHandler {
  /** Plugin registration. */
  public static void registerWith(Registrar registrar) {
    final MethodChannel channel = new MethodChannel(registrar.messenger(), "auro_avatar");
    channel.setMethodCallHandler(new AuroAvatarPlugin());
  }

  @Override
  public void onMethodCall(MethodCall call, Result result) {
    if (call.method.equals("getPlatformVersion")) {
      result.success("Android " + android.os.Build.VERSION.RELEASE);
    } else {
      result.notImplemented();
    }
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'dart:async';

import 'package:flutter/services.dart';
import 'package:auro_avatar/auro_avatar.dart';

void main() => runApp(new MyApp());

class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() => new _MyAppState();
}

class _MyAppState extends State<MyApp> {
  String _platformVersion = 'Unknown';

  @override
  void initState() {
    super.initState();
    initPlatformState();
  }

  // Platform messages are asynchronous, so we initialize in an async method.
  Future<void> initPlatformState() async {
    String platformVersion;
    // Platform messages may fail, so we use a try/catch PlatformException.
    try {
      platformVersion = await AuroAvatar.platformVersion;
    } on PlatformException {
      platformVersion = 'Failed to get platform version.';
    }

    // If the widget was removed from the tree while the asynchronous platform
    // message was in flight, we want to discard the reply rather than calling
    // setState to update our non-existent appearance.
    if (!mounted) return;

    setState(() {
      _platformVersion = platformVersion;
    });
  }

  @override
  Widget build(BuildContext context) {
    return new MaterialApp(
      home: new Scaffold(
        appBar: new AppBar(
          title: const Text('Auro_Avatar plugin example'),
        ),
        body: new Center(
          child: new Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: <Widget>[
              Container(
                child: new InitialNameAvatar(
                  'John Doe',
                  circleAvatar: true,
                  borderColor: Colors.grey,
                  borderSize: 4.0,
                  backgroundColor: Colors.blue,
                  foregroundColor: Colors.white,
                  padding: 20.0,
                  textSize: 30.0,
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
